### PR TITLE
DXCDT-118: Node module e2e test script

### PR DESCRIPTION
**Pull request by willvedd**
## ✏️ Changes

This PR introduces a basic script that runs a primitive end-to-end test of the tool when used as a node module. This script is then integrated into CircleCI and run when appropriate, like before PRs and releases. This script is not meant to test the behavior of the tool, but rather assert that the code compiles and runs to a basic extend; it is meant to help prevent egregious runtime issues. 

**Currently set as a draft while testing with CircleCI.**

## 🔗 References

PR Counterpart: #611 

## 🎯 Testing

This PR does not add application functionality. It does improve testing coverage by ensuring that the CLI works to a minimum expected degree.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,17 +59,20 @@ jobs:
...
           name: test_lts
           v: "lts"
-      - test:
+      - unit_test:
           name: test_current
           v: "18.0.0"
   test_and_deploy:
     jobs:
-      - test:
+      - unit_test:
```
Comment (willvedd) [inline .circleci/config.yml:75]: Renaming to clearly differentiate unit tests from E2E tests.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,8 @@
     "lint:fix": "eslint --fix --ignore-path .eslintignore --ignore-pattern webpack . && kacl lint",
     "lint": "eslint --ignore-path .eslintignore --ignore-pattern webpack . && kacl lint",
     "format": "npx prettier --write .",
-    "test": "ts-mocha -p tsconfig.json --recursive 'test/**/*.test*' --timeout 5000",
+    "test": "ts-mocha -p tsconfig.json --recursive 'test/**/*.test*' --exclude test/e2e/* --timeout 5000",
```
Comment (willvedd) [inline package.json:13]: Repurposing the existing `test` script to be our dedicated unit testing script, which from my experience, has been common. However, it is necessary to exclude the test inside the `test/e2e` directory.

Alternatively, we could re-organize these test in another fashion.

File: package.json [hunk 3]
```diff
@@ -46,6 +47,7 @@
     "sanitize-filename": "^1.6.1",
     "sinon": "^13.0.1",
     "sinon-chai": "^3.7.0",
+    "ts-node": "^10.9.1",
```
Comment (willvedd) [inline package.json:50]: Adding the `ts-node` dependency simplifies the running of the included script by removing the need to transpile the Typescript code into Javascript.

**Review by Widcket — CHANGES_REQUESTED** [review]
Since this requires a secret, make sure to not run it for PRs from forks.

**Comment by Widcket** [comment]
Example: https://github.com/auth0/auth0-flutter/blob/main/.circleci/config.yml#L143

**Comment by willvedd** [comment]
>Since this requires a secret, make sure to not run it for PRs from forks.

@Widcket "Pass secrets to builds from forked pull requests" CircleCI setting disabled.

**Comment by Widcket** [comment]
In that case, the CI will still run the other jobs as expected? Or will the pipeline fail?

**Comment by willvedd** [comment]
>In that case, the CI will still run the other jobs as expected? Or will the pipeline fail?

@Widcket Unclear. My guess is that all jobs will run but overall the pipeline will fail since the e2e jobs will fail when secrets are not provided. A simple solution may be to use CircleCI filters to exclude forks, but not sure if that is an available filter.

**Comment by Widcket** [comment]
> Example: https://github.com/auth0/auth0-flutter/blob/main/.circleci/config.yml#L143

As mentioned above, there's an example on how to avoid running the tests for PRs from forks and not fail the pipeline.

**Review by Widcket — APPROVED** [review]
Good work 💪🏼

**Pull request merged**